lvm2: 2.02.177 -> 2.03.01

markuskowa opened the pull request:
###### Motivation for this change
Version bump. Switched the source to the git repository. Should now be fit for the auto updater.

CC @7c6f434c 

###### Things done
* switch to sources to git
* add `libaio` to `buildInputs`
* Fix grub (fallout from xfsprogs-4.19 update).

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

markuskowa committed:
lvm2: 2.02.177 -> 2.03.01

* switch to sources to git
* add libaio to buildInputs

7c6f434c commented:
@GrahamcOfBorg test installer.lvm

7c6f434c commented:
@GrahamcOfBorg biuld nixosTests.installer.lvm

7c6f434c commented:
And this is the first bump across what they called a set of large refactorings on the mailing list, so probably we shouldn't hurry with backporting?

markuskowa commented:
What do you mean by backporting?
@GrahamcOfBorg build nixosTests.installer.lvm

7c6f434c commented:
I meant `release-18.09`

markuskowa commented:
OK, if there is any need for backporting, there is also lvm2-2.02.183.

7c6f434c commented:
I have some doubts even about 2.02.183

markuskowa commented:
IIRC there were also other problems with the installer tests, such as randomly timed out tests.

7c6f434c commented:
@pbogdan Well, there was also that `spice` failure breaking all the tests…

markuskowa commented:
I just did a rebase. This should eliminate the issue with `spice` and leave only the xfsprogs/grub-os-prober problem on x86_64-linux.
@GrahamcOfBorg build nixosTests.installer.lvm

markuskowa commented:
Now the test times out on x86_64-linux.

7c6f434c commented:
Locally it failed for GRUB reasons without this PR. #51790

markuskowa committed:
grub: add patch for XFS sparse inode feature

xfsprogs-4.16 introduced an new on disk format feature
that is not recognized properly by grub. This patch
allows grub to access XFS filesystem created with
xfsprogs >= 4.16.
This upstream patch can be removed for grub-2.03.

markuskowa commented:
With the XFS patch the lvm installer test succeeds again (locally, with ofborg it times out).

7c6f434c commented on pkgs/tools/misc/grub/2.0x.nix:
```diff
@@ -82,7 +82,16 @@ stdenv.mkDerivation rec {
+    # feature introduced by xfsprogs-4.16.
+    # to be removed in grub-2.03
+    (fetchpatch {
+      url = http://git.savannah.gnu.org/cgit/grub.git/patch/?id=cda0a857dd7a27cd5d621747464bfe71e8727fff;
```
Doesn't it work as https:// ?

markuskowa replied:
Yes, works.

hedning commented:
Test works locally here too :+1: any reason not to merge? at the very worse we'll break an already broke trunk :)

markuskowa commented:
If there are no further comments it can be merged.

hedning commented:
Hmm, seems like `2.03.00` removed  applib, the flag is no longer [recognized](https://hydra.nixos.org/build/85820869/nixlog/1):
> configure: WARNING: unrecognized options: --disable-static, --enable-applib

The lvm2 [commit](https://sourceware.org/git/?p=lvm2.git;a=commit;h=b67ef90438052d3e01f563cdd25e2268f4e02c20) removing it.

[osquery](https://hydra.nixos.org/build/85820869) breaks due to the removal:
> .../block_devices.cpp:24:10: fatal error: lvm2app.h: No such file or directory

`osquery` master [still depends on this](https://github.com/facebook/osquery/blob/master/osquery/tables/system/linux/block_devices.cpp#L24), so it's not enough to upgrade `osquery`.

cc latest `osquery` maintainer @Ma27

markuskowa commented:
The lvm update also breaks `udisks1` and `udisks-glue`. The latest versions of those packages are from 2013 and 2014 and look rather outdated. Can they be removed (CC @pSub)?

Ma27 commented:
Even if it would be fixed on `oquery` master, it would take some time until I do the next `osquery` bump, unfortunately the diff between the patched cmake (I'm partially responsible for) and master's cmake is way too big now.

Until this is fixed upstream and I do the upgrade for `nixpkgs`, wouldn't it be possible to maintain two versions of the package (as this seems to be a breaking change)?

7c6f434c commented:
Reinstating 2.02.183 seems to be a natural plan…

markuskowa commented:
So far it looks like  `osquery` is the only package not ready lvm2-2.03.x. Makes sense to add 2.02.183 as a legacy version. We can keep a legacy version until `osquery` is updated.
